File: 0014-longest-common-prefix/0014-longest-common-prefix.go
import (
    "strings"
)

func longestCommonPrefix(strs []string) string {
    var longestPrefix strings.Builder
    for i := 0; i < len(strs[0]); i++ {
        ch := strs[0][i]
        for _, s := range strs[1:] {
            if i >= len(s) || s[i] != ch {
                return longestPrefix.String()
            }
        }
        longestPrefix.WriteByte(ch)
    }
    
    return longestPrefix.String()
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        longest = ''
        for i in range(len(strs[0])):
            char = strs[0][i]
            for s in strs[1:]:
                if i >= len(s) or s[i] != char:
                    return longest
                
            longest += char
                
        return longest
                